lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -113,7 +113,10 @@ def process_resource(serializer, primary)
         end
 
         def process_relationships(serializer, include_tree)
+          resource_identifier    = ApiObjects::ResourceIdentifier.new(serializer).as_json
+          requested_associations = fieldset.fields_for(resource_identifier[:type])
           serializer.associations(include_tree).each do |association|
+            next unless requested_associations.nil? || requested_associations.include?(association.name)
```

Comment:
Probably not the most beautiful solution but it is already done in a similar way in the [Attributes](https://github.com/rails-api/active_model_serializers/blob/master/lib/active_model/serializer/attributes.rb#L21) class and finding a cleaner solution would have implied too much refactoring for this small fix.